src/app/component/allresults/allresults.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { QuestionService } from '../../service/question.service';
+
+@Component({
+  selector: 'app-allresults',
+  templateUrl: './allresults.component.html',
+  styleUrls: ['./allresults.component.css'],
+})
+export class AllresultsComponent implements OnInit, OnDestroy {
+  userName: any;
+  initialData: number = 8;
+  allResultData: any[] | undefined;
+
+  constructor(public questionService: QuestionService) {}
+
+  ngOnInit(): void {
+    this.resultData();
+    if (!this.userName) {
+      this.getData();
+    }
+  }
+
+  getData() {
+    let data: any = localStorage.getItem('registeruser');
+    this.userName = JSON.parse(data).find((data: any) => {
+      return data.id == this.questionService.getUser();
+    })?.fullname;
+  }
+
+  resultData() {
+    let data: any = localStorage.getItem('result');
+    this.allResultData = JSON.parse(data).reverse();
+  }
+
+  initialLetter(userName: string) {
```

Comment:
Please change this function name to something like getUserLetter

Reply:
Yes i changed the function name